### PR TITLE
Add chinese translation for Book chapter 8 machine translation

### PR DESCRIPTION
**Pull request by nickyfantasy**

### CODE REVIEW

File: 08.machine_translation/README.cn.md [hunk 1]
```diff
@@ -165,338 +127,316 @@ $$e_{ij} = {align(z_i, h_j)}$$
...
+import numpy as np
+import paddle
+import paddle.fluid as fluid
+import paddle.fluid.framework as framework
+import paddle.fluid.layers as pd
+from paddle.fluid.executor import Executor
+from functools import partial
+import os
+
+dict_size = 30000
```
Comment (jetfuel) [inline 08.machine_translation/README.cn.md:146]: Please keep the original comments. They are quite helpful for user to under.

File: 08.machine_translation/README.cn.md [hunk 2]
```diff
@@ -165,338 +127,316 @@ $$e_{ij} = {align(z_i, h_j)}$$
...
-    target_dict_dim = dict_size # 目标语言字典维度
-    word_vector_dim = 512 # 词向量维度
-    encoder_size = 512 # 编码器中的GRU隐层大小
-    decoder_size = 512 # 解码器中的GRU隐层大小
-    beam_size = 3 # 柱宽度
-    max_length = 250 # 生成句子的最大长度
-    ```
+   ```python
+   def encoder(is_sparse):
+    # encoder
```
Comment (jetfuel) [inline 08.machine_translation/README.cn.md]: No need for this comment.

File: 08.machine_translation/README.cn.md [hunk 3]
```diff
@@ -165,338 +127,316 @@ $$e_{ij} = {align(z_i, h_j)}$$
...
+    return encoder_out
+   ```
 
-2. 其次，实现编码器框架。分为三步：
+再实现训练模式下的解码器：
 
-   - 输入是一个文字序列，被表示成整型的序列。序列中每个元素是文字在字典中的索引。所以，我们定义数据层的数据类型为`integer_value_sequence`（整型序列），序列中每个元素的范围是`[0, source_dict_dim)`。
+```python
+   def train_decoder(context, is_sparse):
+    # decoder
```
Comment (jetfuel) [inline 08.machine_translation/README.cn.md]: No need for this.

File: 08.machine_translation/README.cn.md [hunk 4]
```diff
@@ -165,338 +127,316 @@ $$e_{ij} = {align(z_i, h_j)}$$
...
-      - context通过调用`simple_attention`函数，实现公式$c_i=\sum {j=1}^{T}a_{ij}h_j$。其中，enc_vec是$h_j$，enc_proj是$h_j$的映射（见3.1），权重$a_{ij}$的计算已经封装在`simple_attention`函数中。
-      - decoder_inputs融合了$c_i$和当前目标词current_word（即$u_i$）的表示。
-      - gru_step通过调用`gru_step_layer`函数，在decoder_inputs和decoder_mem上做了激活操作，即实现公式$z_{i+1}=\phi _{\theta '}\left ( c_i,u_i,z_i \right )$。
-      - 最后，使用softmax归一化计算单词的概率，将out结果返回，即实现公式$p\left ( u_i|u_{&lt;i},\mathbf{x} \right )=softmax(W_sz_i+b_z)$。
+```python
+train_reader = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.wmt14.train(dict_size), buf_size=1000),
+        batch_size=batch_size)
+```
```
Comment (jetfuel) [inline 08.machine_translation/README.cn.md:313]: Can we show some example outputs during training? Should be helpful.

**Review by jetfuel — APPROVED** [review]
I don't see we save the parameter anywhere. How does this network re-use its trained parameters?

Otherwise LGTM.

**Review by daming-lu — APPROVED** [review]

**Pull request merged**